### ISSUE TITLE
Invalid zone setting "**" (value: ) found - cannot be set as it is read only

### ISSUE THREAD

**Issue by davidvpe** [cloudflare/terraform-provider-cloudflare#1498]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v3.10.0
+ provider registry.terraform.io/hashicorp/aws v3.74.0

### Affected resource(s)

- cloudflare_zone_settings_override

### Terraform configuration files

```hcl
data "cloudflare_zone" "zone" {
  name = "example.org"
}

resource "cloudflare_zone_settings_override" "settings" {
  zone_id  = data.cloudflare_zone.zone.id

  settings {
    ssl                         = "full"
    always_use_https            = "on"
  }
}
```

### Debug output

output too long...

### Panic output

**Error: invalid zone setting "polish" (value: ) found - cannot be set as it is read only**

### Expected output

All changes applied

### Actual output

**Error: invalid zone setting "polish" (value: ) found - cannot be set as it is read only**

and every time, the setting that causes the error is different (it just throws a random readonly setting every time), but as you can see on the example above, I am never setting it.

### Steps to reproduce

1. Create a zone in Cloudflare
2. Reference that zone as `data` on your main.tf
3. Then add a resource `cloudflare_zone_settings_override` and use the zone_id from the `data` from step 2, after this, add the `settings` block and set one setting such as `ssl = "full"`
4. Try to apply the changes

### Additional factoids

This happens on both local and latest stable docker image on Gitlab CI 

### References

Duplicate of #1117

**Comment by davidvpe** [comment]
For people who is having this same issue try the following:

1. Checkout your state file (in case you have it remotely like me), I recommend you switch your backend to a local one (you'll have to do a migration).
2. Delete the settings_override resource from your state file and also from your tf file
3. Do a `terraform apply`
4. Now re-add the settings_override resource
5. Do a `terraform apply` (It should work now!)
6. Now change your backend back to what it was and do the migration in case needed

I'm not really sure what caused this, but in my case it was because I did a terraform apply, and the script failed while being created (I did the terraform validate, and plan but no errors were thrown).
